airflow/contrib/hooks/gcp_container_hook.py

```diff
@@ -0,0 +1,215 @@
+        :type timeout: float
+        :return: The full url to the delete operation if successful, else None
+        """
+
+        self.log.info("Deleting (project_id={}, zone={}, cluster_id={})".format(
+            self.project_id, self.location, name))
+
+        try:
+            op = self.client.delete_cluster(project_id=self.project_id,
+                                            zone=self.location,
+                                            cluster_id=name,
+                                            retry=retry,
+                                            timeout=timeout)
+            op = self.wait_for_operation(op)
+            # Returns server-defined url for the resource
+            return op.self_link
+        except exceptions.NotFound as error:
+            self.log.info('Assuming Success: ' + error.message)
+
+    def create_cluster(self, cluster, retry=DEFAULT, timeout=DEFAULT):
+        """
+        Creates a cluster, consisting of the specified number and type of Google Compute
+        Engine instances.
+
+        :param cluster: A Cluster protobuf or dict. If dict is provided, it must be of
+            the same form as the protobuf message google.cloud.container_v1.types.Cluster
+        :type cluster: dict or google.cloud.container_v1.types.Cluster
+        :param retry: A retry object (google.api_core.retry.Retry) used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The amount of time, in seconds, to wait for the request to
+            complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :return: The full url to the new, or existing, cluster
+        :raises
+            ParseError: On JSON parsing problems when trying to convert dict
+            AirflowException: cluster is not dict type nor Cluster proto type
+        """
+
+        if isinstance(cluster, dict):
+            cluster_proto = Cluster()
+            cluster = self._dict_to_proto(py_dict=cluster, proto=cluster_proto)
+        elif not isinstance(cluster, Cluster):
+            raise AirflowException(
+                "cluster is not instance of Cluster proto or python dict")
+
+        self._append_label(cluster, 'airflow-version', 'v' + version.version)
+
+        self.log.info("Creating (project_id={}, zone={}, cluster_name={})".format(
+            self.project_id,
+            self.location,
+            cluster.name))
+        try:
+            op = self.client.create_cluster(project_id=self.project_id,
+                                            zone=self.location,
+                                            cluster=cluster,
+                                            retry=retry,
+                                            timeout=timeout)
+            op = self.wait_for_operation(op)
+
+            return op.target_link
```

Comment:
nit: op.target_link isn't the same as get_cluster output, good to be consistent.